integreat_cms/deepl_api/utils.py

```diff
@@ -58,84 +112,117 @@ def deepl_translation(self, request, content_objects, language_slug, form_class)
         :type form_class: ~integreat_cms.cms.forms.custom_content_model_form.CustomContentModelForm
         """
-        # Get target language
-        target_language = request.region.get_language_or_404(language_slug)
-        source_language = request.region.get_source_language(language_slug)
-        for content_object in content_objects:
-            source_translation = content_object.get_translation(source_language.slug)
-            if not source_translation:
-                messages.error(
-                    request,
-                    _('No source translation could be found for {} "{}".').format(
-                        type(content_object)._meta.verbose_name.title(),
-                        content_object.best_translation.title,
-                    ),
-                )
-                continue
-            existing_target_translation = content_object.get_translation(
-                target_language.slug
-            )
-            # For some languages, the DeepL client expects the BCP tag instead of the short language code
-            if target_language.slug in ("en", "pt"):
-                target_language_key = target_language.bcp47_tag
-            else:
-                target_language_key = target_language.slug
-            target_title = self.translator.translate_text(
-                source_translation.title,
-                source_lang=source_language.slug,
-                target_lang=target_language_key,
-            )
-            data = {
-                "title": target_title,
-                "status": existing_target_translation.status
-                if existing_target_translation
-                else source_translation.status,
-            }
-            if source_translation.content:
-                data["content"] = self.translator.translate_text(
-                    source_translation.content,
-                    source_lang=source_language.slug,
-                    target_lang=target_language_key,
-                )
-            # for pois adds a meta description
-            if hasattr(source_translation, "meta_description"):
-                data["meta_description"] = self.translator.translate_text(
-                    source_translation.meta_description,
-                    source_lang=source_language.slug,
-                    target_lang=target_language_key,
-                )
-            content_translation_form = form_class(
-                data=data,
-                instance=existing_target_translation,
-                additional_instance_attributes={
-                    "creator": request.user,
-                    "language": target_language,
-                    source_translation.foreign_field(): content_object,
-                },
+        # Re-select the region from db to prevent simultaneous
+        # requests exceeding the DeepL usage limit
+
+        with transaction.atomic():
+            region = (
+                apps.get_model("cms", "Region")
```

Comment:
I think this is fine! I see no way around it since the `DeepLApi` class is imported in `deepl_api`'s `apps.py` which has a [`ready()`](https://github.com/digitalfabrik/integreat-cms/blob/ad73c79b7b994b756997fb1019b63662345ff543/integreat_cms/deepl_api/apps.py#L27)-function which is run when the deepl api app is initialized.
So my theory why the `AppRegistryNotReady` error appears is that the `ready()`-function is executed before the models are loaded, meaning we cannot import them at this point in time.

Reply:
Thanks for the explanation. After asking I realized the workaround with `get_model` was used in lots of places, I guess this has happened before? :D